src/Unleash/Scheduling/FetchFeatureTogglesTask.cs

```diff
@@ -28,20 +31,32 @@ internal class FetchFeatureTogglesTask : IUnleashScheduledTask
             ThreadSafeToggleCollection toggleCollection,
             IJsonSerializer jsonSerializer,
             IFileSystem fileSystem,
+            EventCallbackConfig eventConfig,
             string toggleFile,
             string etagFile)
         {
             this.apiClient = apiClient;
             this.toggleCollection = toggleCollection;
             this.jsonSerializer = jsonSerializer;
             this.fileSystem = fileSystem;
+            this.eventConfig = eventConfig;
             this.toggleFile = toggleFile;
             this.etagFile = etagFile;
         }
 
         public async Task ExecuteAsync(CancellationToken cancellationToken)
         {
-            var result = await apiClient.FetchToggles(Etag, cancellationToken).ConfigureAwait(false);
+            FetchTogglesResult result;
+            try
+            {
+                result = await apiClient.FetchToggles(Etag, cancellationToken).ConfigureAwait(false);
```

Comment:
Looks like this call already raises an event, what does this catch block now do?

Reply:
Catches very specific exceptions and let the rest bubble up as before

Reply:
Ah, gotcha happens when http fails and the other event shouldn't be raised when that happens - sounds sane